realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -497,9 +511,15 @@ private void completedAsyncRealmObject(QueryUpdateTask.Result result) {
                     realmObject.onCompleted(rowPointer);
                     realmObject.notifyChangeListeners();
 
-                } else if (compare > 0) {
-                    // the caller has advanced we need to
-                    // retry against the current version of the caller if it's still empty
+                } else {
+                    if (compare > 0) {
+                        RealmLog.d("[COMPLETED_ASYNC_REALM_OBJECT "+ realmObject + "] , realm:" + HandlerController.this
```

Comment:
No? `Row.EMPTY_ROW.isAttached()` should return `false`. Otherwise, the following line (523) should check `isLoaded` instead?